feat(gcp_stackdriver_logs sink): make project, folder, org and billing_account templateable (#22459)
vectordotdev/vector: changelog.d/22433_templateable_stackdriver_log_name.feature.md

> Sink `gcp_stackdriver_logs` now supports templateable destinations via `project_id`, `folder_id`, `organization` and `billing_account_id`.

Reviewer: ```suggestion
`project_id`, `folder_id`, `organization`, and `billing_account_id` fields in the `gcp_stackdriver_logs` sink are now templateable.
```

Author: Applied. thanks
